### PR TITLE
fix(selectionLimit): 🐛 use ACTION_GET_CONTENT for multiple image selection instead of ACTION_PICK

### LINKED ISSUE #1739: [🐛] Multiple image selection not working on Samsung devices.
### Description

Setting selectionLimit has no effect on file selection on Samsung devices when choosing the Samsung Gallery App. Works as expected in iOS, Google photos and every other brand gallery.

### How to repeat issue and example

- Call launchImageLibrary({mediaType: 'photo', selectionLimit: 0}) on a Samsung device
- Choose Gallery when prompted.
- Try to select multiple files.

### Solution

Behaviour should be no different from other brands of devices.

### Additional Information

- Image Picker version: 4.0.3
- React Native version: 0.64.0
- Platform: Android
- Development Operating System: MacOS Big Sur 11.4
- Dev tools: Tested on Samsung Galaxy S9+ running Android 10 / One UI 2.5 and Samsung A20 Running Android 10 / One UI 2.0

### PR DESCRIPTION
**Pull request by Johan-dutoit**
Should close #1739

### CODE REVIEW

File: android/src/main/java/com/imagepicker/ImagePickerModule.java [hunk 1]
```diff
@@ -127,21 +127,27 @@ public void launchImageLibrary(final ReadableMap options, final Callback callbac
...
-            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*", "video/*"});
             libraryIntent.addCategory(Intent.CATEGORY_OPENABLE);
+            libraryIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
         }
 
-        if (this.options.selectionLimit != 1) {
-            libraryIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+        if(isPhoto) {
+            libraryIntent.setType("image/*");
+            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*"});
```
Comment (ravirajn22) [inline android/src/main/java/com/imagepicker/ImagePickerModule.java:144]: `libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*"}); is not required when u setType. The Mime type we use in setType will be used.

File: android/src/main/java/com/imagepicker/ImagePickerModule.java [hunk 2]
```diff
@@ -127,21 +127,27 @@ public void launchImageLibrary(final ReadableMap options, final Callback callbac
...
+        } else if (isVideo) {
+            libraryIntent.setType("video/*");
+            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"video/*"});
```
Comment (ravirajn22) [inline android/src/main/java/com/imagepicker/ImagePickerModule.java:147]: Same I like above

**Comment by Johan-dutoit** [comment]
@ravirajn22 changes made.

File: android/src/main/java/com/imagepicker/ImagePickerModule.java [hunk 3]
```diff
@@ -127,21 +127,27 @@ public void launchImageLibrary(final ReadableMap options, final Callback callbac
...
+
+        if(isSingleSelect && (isPhoto || isVideo)) {
             libraryIntent = new Intent(Intent.ACTION_PICK);
-            libraryIntent.setType("image/*");
         } else {
             libraryIntent = new Intent(Intent.ACTION_GET_CONTENT);
-            libraryIntent.setType("*/*");
-            libraryIntent.putExtra(Intent.EXTRA_MIME_TYPES, new String[]{"image/*", "video/*"});
             libraryIntent.addCategory(Intent.CATEGORY_OPENABLE);
+            libraryIntent.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
```
[the remark this answers is not in the document]
Comment (Johan-dutoit) [reply android/src/main/java/com/imagepicker/ImagePickerModule.java:139]: Not with the if statement above it

[the remark this answers is not in the document]
Comment (Johan-dutoit) [reply android/src/main/java/com/imagepicker/ImagePickerModule.java:142]: Sorted

**Comment by ravirajn22** [comment]
LGTM, i am unable to test. If you can, manually test ones, merge it.

**Comment by axeljeremy7** [comment]
Do I have to add this `  <action android:name="android.intent.action.ACTION_GET_CONTENT"/>` ?

**Pull request merged**